packages/teleterm/src/services/config/configService.ts

```diff
@@ -1,39 +1,203 @@
-import { merge } from 'lodash';
+import { z } from 'zod';
 
-import { keyboardShortcutsConfigProvider } from './providers/keyboardShortcutsConfigProvider';
-import { appearanceConfigProvider } from './providers/appearanceConfigProvider';
-import { Config, ConfigService, ConfigServiceProvider } from './types';
+import { FileStorage } from 'teleterm/services/fileStorage';
+import { Platform } from 'teleterm/mainProcess/types';
 
-type ConfigServiceProviders = {
-  readonly [Property in keyof Config]: ConfigServiceProvider<Config[Property]>;
+import { createConfigStore } from './configStore';
+
+const createAppConfigSchema = (platform: Platform) => {
+  const defaultKeymap = getDefaultKeymap(platform);
+  const defaultFonts = getDefaultFonts(platform);
+
+  // Important: all keys except 'usageMetrics.enabled' are currently not
+  // configurable by the user, we need to better validate the input (especially
+  // fonts, where any CSS can be injected).
```

Comment:
```suggestion
  // configurable by the user. Before we let the user configure them,
  // we need to set up some actual validation, so that for example
  // arbitrary CSS cannot be injected into the app through font settings.
  //
  // However, we want them to be in the config schema, so we included
  // them here but we do not read their value from the stored config.
```